### DISCUSSION TITLE
레벨4 무엇을 할까요?

### DISCUSSION THREAD

**Discussion by WaiNaat** [woowacourse-teams/2023-pium#317] [개발 논의]
- 기능 고도화 vs 성능 고도화 vs 새로운 기능
    - 5차, 6차 데모에서 기능 요구사항보다는 기술 요구사항이 많음
    - 꼭 실현가능한 기능이 아니더라도 정했으면 좋을 것 같다

### 커뮤니티

사용자 입장에서는 내가 키우는 식물을 남들은 어떻게 키우는지 궁금할 수밖에 없다.
내 식물을 자랑하고 싶은 마음

현 피움 서비스는 도구: 나의 정보를 다른 어딘가에 저장하는 느낌.
인터넷의 힘은 사람과 사람의 연결이기에 그러한 강점을 활용하지 못한다는 생각이 든다. 단순히 게시판이 있는 커뮤니티로 하면 잡담같은 게 많을 수 있어서 도메인에 관련된 것으로 좁힐 필요성이 있다.

1. 비슷한 환경을 가진 식물들에 대한 기록을 공유
나는 이러한 환경에서 이렇게 키우고 있는데 남들은 어떻게 하지? 라는 궁금증을 해결할 수 있음. 오래 키운 사람의 기록을 참고해서 나도 배울 수 있고. 아니면 실패 기록을 통해서 배울 수도 있다.
이 방법의 문제점은 기록이 별로 없는 지금은 큰 의미가 없다는거

2. 물주기 기록, 환경 기록 등 기록을 공유할 수 있는 기능 (일종의 타임라인 공유)
기록을 제공해야 한다는 것 자체는 동의하지만, 어떤 방식으로 보여줄 것인지를 잘 정해야 할 것 같다. 500일의 기록이 담긴 타임라인? 보기 너무 힘들다.

3. 통계를 이용한 정보 제공
통계를 이미지로 뽑아주는 기능을 통해 인스타 등으로 바로 공유 가능하게끔 하는 방식도 있다.
사전 식물 정보에서 통계를 보여주면 더 좋을 것 같다.
내 관리법을 등록해서 다른 사람들이 사전 정보에서 찾아볼 수 있는 게시판 형태

4. 사용자가 관리법을 공유하는 방식
일종의 짧은 리뷰를 공유

6. 댓글을 통한 질의응답
관리자 페이지를 만들어야 함

**통계의 신뢰성과 통계 로직 구현의 어려움 vs 댓글형 커뮤니티의 관리 어려움**

### 반려식물 사진 등록

커뮤니티성 기능 이전에 이게 필요할 것 같다.

### 식물이 없어요 (등록 신청)

사용자가 임의로 등록할 수 있게 하면 문제가 생길 수 있음
사용자가 신청서를 작성하는 방식 (검색해서 나오지 않았을 경우 신청하기 버튼을 통해 관리자에게 신청)
신청서에서 식물명(또는 몰라요)과 사진(선택)을 받아서 검수 후 등록

등록 이후 사용자에게 알림 또는 공지사항을 통해서 전달 (추후 논의)

### 우선순위

1. 반려 식물 사진 등록 ( @hozzijeong, @Kim0914 )
  - 입력 폼에서 등록 1장
  - 수정 가능
  - 삭제 가능 (삭제 시 기본이미지는 사전 식물의 이미지)
2. 관리자 페이지 ( @Choi-JJunho )
3. 사전 식물 등록 요청 ( @WaiNaat, @rawfishthelgh )
  - 검색 결과가 없으면 등록신청 버튼 뿅!
4. 반려 식물 XX 정보 공유 (디자인이 나와야하는 작업) ( @bassyu , @yeonkkk )
  - 동작 방식은 다음과 같다.
    - 공유 버튼을 누르면
    - -> 코멘트 작성 폼이 뿅! 
    - -> 코멘트 + 난이도(모름, 쉬움, 보통, 어려움) 를 입력한다.
    - -> 반려식물 기본정보 + 작성한 내용 기반으로 리뷰가 작성된다.
  - 해당 정보는 새로운 Form으로 작성된다. (난이도 + 코멘트 만 들어간다. PetPlant 정보는 기존 정보를 가져온다.)
5. 신고하기 기능 ( )
6. 마이페이지 고도화
  - 사용자 닉네임 추가 (후순위, 자체 닉네임 테이블)
  - 공유한 리뷰 모아보기

**Comment by Choi-JJunho** [comment]
개발 완료
- https://github.com/woowacourse-teams/2023-pium/discussions/386
- [Pium 1.1.0🌱 Release Note](https://github.com/woowacourse-teams/2023-pium/releases/tag/v1.1.0)
